packages/upload-api/src/provider-add.js

```diff
@@ -1,25 +1,21 @@
-import * as API from '../api.js'
+import * as API from './types.js'
 import * as Server from '@ucanto/server'
 import { Provider } from '@web3-storage/capabilities'
 import * as validator from '@ucanto/validator'
 
 /**
- * @param {Context} ctx
+ * @param {API.ProviderServiceContext} ctx
  */
 export const provide = (ctx) =>
   Server.provide(Provider.add, (input) => add(input, ctx))
 
 /**
- * @typedef {object} Context
- * @property {object} models
- * @property {API.ProvisionsStorage} models.provisions
- *
  * @param {API.Input<Provider.add>} input
- * @param {Context} context
+ * @param {API.ProviderServiceContext} context
  */
 export const add = async (
   { capability, invocation },
-  { models: { provisions } }
+  { provisionsStorage: provisions }
 ) => {
   const {
```

Comment:
Does not need to be part of this PR, but I would really like if this instead invoked `customer.add` from the service DID, and then invoked `consumer.add` also from the service DID.

That is conceptually what supposed to be happening and provider/add is simply a sugar to save roundtrip.

Reply:
yep agreed. also think we shouldn't get this in in this PR, but I'd like to move in this direction as well.